etl-pipeline/processes/pdf_generation/pdf_generate.py

```diff
@@ -20,103 +28,189 @@
 logger = create_log(__name__)
 
 
-class PDFGenerationProcess:
-    def __init__(self, bucket, ocr_dir, mets_file):
-        self.bucket_name = bucket
-        self.bucket = s3.Bucket(self.bucket_name)
-        self.ocr_dir = ocr_dir
-        self.mets = mets_file
+def generate_pdf(
+    bucket_name: str, barcode: str, ocr_dir: str, mets_file_key: str
+) -> dict:
+    bucket = s3.Bucket(bucket_name)
 
-    def run_process(self):
-        mets_file = mets_parser.METSFile.from_mets_str(
-            self.bucket.get(key=self.mets)["Body"].read(),
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        mets_file, mets_path, metadata, xml_data = _read_mets_and_metadata(
+            bucket, mets_file_key
         )
-        mets_path = path.METSPath(self.mets)
 
-        metadata = model.get_metadata(self.bucket, mets_path, mets_file)
-        model.write_metadata(self.bucket, mets_path, metadata, mets_path.tagged_pdf_key)
+        record, is_public_domain = _map_record_and_rights(xml_data)
+        if is_public_domain:
+            upload_bucket_name = os.environ["FILE_BUCKET"]
+        else:
+            upload_bucket_name = os.environ["PRIVATE_FILE_BUCKET"]
+        upload_bucket = s3.Bucket(upload_bucket_name)
```

Comment:
i think we can come back to this in a separate PR but not sure if this s3 file and Bucket class will have the right acl configuration to upload the pdf to be publicly downloadable.